src/tactic/solve_by_elim.lean

```diff
@@ -164,19 +164,27 @@ open solve_by_elim
 /--
 `solve_by_elim` repeatedly tries `apply`ing a lemma
 from the list of assumptions (passed via the `opt` argument),
-recursively operating on any generated subgoals.
+recursively operating on any generated subgoals, backtracking as necessary.
 
-It succeeds only if it discharges the first goal
-(or with `backtrack_all_goals := tt`, if it discharges all goals.)
+`solve_by_elim` succeeds only if it discharges the goal.
+(By default, `solve_by_elim` focuses on the first goal, and only attempts to solve that.
+With the option `backtrack_all_goals := tt`,
+it attempts to solve all goals, and only succeeds if it does so.
+With `backtrack_all_goals := tt`, `solve_by_elim` will backtrack a solution it has found for
+one goal if it then can't discharge other goals.)
```

Comment:
Should some or all of this info be in the docstring for the `interactive` version too?